src/main/java/io/vertx/ext/stomp/impl/Subscription.java

```diff
@@ -0,0 +1,145 @@
+package io.vertx.ext.stomp.impl;
+
+import io.vertx.ext.stomp.Frame;
+import io.vertx.ext.stomp.StompServerConnection;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents a subscription.
+ * This class is thread safe.
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+public class Subscription {
```

Comment:
This is because the server can implement a custom behavior. It cannot be done directly in the onAck and onNack handlers as they just receive a single frame, not the whole set of (not-)acknowledged frames.